### PR TITLE
Ear 234 collapsed sections show errors

### PR DESCRIPTION
**Pull request by zimboflyman**
### What is the context of this PR?

This is to show if validation errors exist within a sections Questions, when the section accordion is closed - in left hand nav

*Acceptance criteria (user perspective)*
GIVEN I'm on a section that has validation errors contained **_within it or on it_**
WHEN the section is collapsed in the left panel
THEN a red circle will appear alongside it in the left panel

### How to review 

Create a section with an answer that has at least one validation error.
The section should NOT have a validation error,
Close the section accordion
a red error badge should appear,
open the section
The badge should move to the corresponding answer.

If the section has it's own validation error:
Then that error badge will show - regardless if the contained answers have errors or not.

There will be no error count in the section error badge
only in the answer error badges

1. * [ ] Bring this branch up-to-date with Origin/Master
2. * [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. * [ ] Click the **Merge** button on this pull request
4. * [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. * [ ] Move the Trello card for this task into the next stage of the process

### CODE REVIEW

File: eq-author/src/App/QuestionnaireDesignPage/NavigationSidebar/SectionNavItem.js [hunk 1]
```diff
@@ -43,18 +43,38 @@ export class UnwrappedSectionNavItem extends React.Component {
...
       tab: match.params.tab,
     });
 
-    const SectionTitle = () => (
+    let questionErrorCount = 0;
+
+    // section.pages.forEach(val => {
+    //   questionErrorCount =
+    //     questionErrorCount + section.pages.validationErrorInfo.totalCount;
+    // });
```
Comment (tjamesmac) [inline eq-author/src/App/QuestionnaireDesignPage/NavigationSidebar/SectionNavItem.js:69]: Just a few comments

File: eq-author/src/App/QuestionnaireDesignPage/NavigationSidebar/SectionNavItem.js [hunk 2]
```diff
@@ -43,18 +43,38 @@ export class UnwrappedSectionNavItem extends React.Component {
     questionnaire: CustomPropTypes.questionnaire,
     section: CustomPropTypes.section.isRequired,
     match: CustomPropTypes.match.isRequired,
+    //  sectionErrors: PropTypes.number.isRequired,
```
Comment (MattElcock) [inline eq-author/src/App/QuestionnaireDesignPage/NavigationSidebar/SectionNavItem.js:46]: Seems like a comment has been left in by mistake.

File: eq-author/src/App/QuestionnaireDesignPage/NavigationSidebar/SectionNavItem.js [hunk 3]
```diff
@@ -43,18 +43,38 @@ export class UnwrappedSectionNavItem extends React.Component {
...
   };
 
   render() {
-    const { questionnaire, section, match, ...otherProps } = this.props;
+    const {
+      questionnaire,
+      section,
+      match,
+      //  sectionErrors,
```
Comment (MattElcock) [inline eq-author/src/App/QuestionnaireDesignPage/NavigationSidebar/SectionNavItem.js:54]: Seems like a comment has been left in by mistake.

File: eq-author/src/App/QuestionnaireDesignPage/NavigationSidebar/SectionNavItem.js (hunk 1 above)
Comment (zimboflyman) [reply eq-author/src/App/QuestionnaireDesignPage/NavigationSidebar/SectionNavItem.js:69]: bugger - my worst habit!

**Pull request merged**